Last-Server-In-Cache Group validation runs when Cache Group has not changed

<!--
************ STOP!! ************
If this issue identifies a security vulnerability, DO NOT submit it! Instead, contact
the Apache Traffic Control Security Team at security@trafficcontrol.apache.org and follow the
guidelines at https://www.apache.org/security/ regarding vulnerability disclosure.

- For *SUPPORT QUESTIONS*, use the Traffic Control slack (https://s.apache.org/atc-slack)
or Traffic Control mailing lists (https://trafficcontrol.apache.org/mailing_lists).
- Before submitting, please **SEARCH GITHUB** for a similar issue or PR.
-->

## I'm submitting a ...
<!-- delete all those that don't apply -->
<!--- security vulnerability (STOP!! - see above)-->
-  bug report

## Traffic Control components affected ...
<!-- delete all those that don't apply -->
-  Traffic Ops

## Current behavior:
<!-- Describe how the bug manifests -->
Server validation will fail due to it being the last Server in its Cache Group if
1. The Server is in a Topology-based Cache Group
2. The Server is updated

## Expected behavior:
<!-- Describe what the behavior would be without the bug -->
Validation should only fail on a Server due to it being the last Server in its Cache Group if the Cache Group of the Server has changed.

## Minimal reproduction of the problem with instructions:
<!--
If the current behavior is a bug, please provide the *STEPS TO REPRODUCE* and
include the applicable TC version.
-->
1. Make a Cache Group
2. Make a Server
3. Add the Server to the Cache Group
4. Make a Topology
5. Add the Server to the Topology
6. Update the Server without changing anything

## Anything else:
<!--
e.g. stacktraces, related issues, suggestions how to fix (feel free to delete
this section)
-->
Found because `UniqueIPProfileTestServers` runs on a random Server each time and was failing 20% of the time due to this bug. Last 12 lines of that:

```go
servers_test.go:296: Response:  midInTopologyEdgeCg06   {[{Server created success}]}
servers_test.go:296: Response:  midInTopologyEdgeCg07   {[{Server created success}]}
topologies_test.go:56: Response:  &{{a multi-site origin topology mso-topology [{0 multiOriginCachegroup [] <nil>} {0 parentCachegroup [0] <nil>} {0 secondaryCachegroup [0] <nil>} {0 cachegroup3 [1 2] <nil>}] <nil>} {[{topology was created. success}]}}
topologies_test.go:56: Response:  &{{another topology another-topology [{0 parentCachegroup [] <nil>} {0 cachegroup1 [0] <nil>} {0 secondaryCachegroup [] <nil>} {0 cachegroup2 [2] <nil>}] <nil>} {[{topology was created. success}]}}
topologies_test.go:56: Response:  &{{A topology with secondary parents secondary-parents [{0 parentCachegroup [] <nil>} {0 cachegroup1 [0 2] <nil>} {0 secondaryCachegroup [] <nil>} {0 fallback1 [2 0] <nil>} {0 fallback2 [2 0] <nil>}] <nil>} {[{topology was created. success}]}}
topologies_test.go:56: Response:  &{{A 4-tier topology 4-tiers [{0 parentCachegroup [] <nil>} {0 parentCachegroup2 [0] <nil>} {0 cachegroup1 [1] <nil>} {0 secondaryCachegroup [1 0] <nil>} {0 fallback1 [3] <nil>}] <nil>} {[{topology was created. success}]}}
topologies_test.go:56: Response:  &{{This topology stems from 2 ancestors forked-topology [{0 topology-edge-cg-01 [2] <nil>} {0 topology-edge-cg-02 [6] <nil>} {0 topology-mid-cg-01 [3] <nil>} {0 topology-mid-cg-02 [4] <nil>} {0 topology-mid-cg-03 [5] <nil>} {0 topology-mid-cg-04 [] <nil>} {0 topology-mid-cg-05 [7] <nil>} {0 topology-mid-cg-06 [8] <nil>} {0 topology-mid-cg-07 [] <nil>}] <nil>} {[{topology was created. success}]}}
topologies_test.go:56: Response:  &{{a topology top-for-ds-req [{0 dtrc1 [] <nil>} {0 dtrc2 [0] <nil>} {0 dtrc3 [0] <nil>}] <nil>} {[{topology was created. success}]}}
topologies_test.go:56: Response:  &{{a topology top-for-ds-req2 [{0 dtrc1 [] <nil>} {0 dtrc2 [0] <nil>}] <nil>} {[{topology was created. success}]}}
servers_test.go:653: expected update to pass: 400 Bad Request[400] - Error requesting Traffic Ops https://localhost:6443/api/3.0/servers/47122 {"alerts":[{"text":"server is the last one in its cachegroup, which is used by a topology, so it cannot be moved to another cachegroup: cachegroups with no servers in them: multiOriginCachegroup in topologies: mso-topology","level":"error"}]}
types_test.go:208: ---- DeleteTestTypes ----
--- FAIL: TestServers (8.35s)
```

<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

    https://apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

--- a/traffic_ops/testing/api/v3/servers_test.go
+++ b/traffic_ops/testing/api/v3/servers_test.go
@@ -82,10 +82,16 @@ func LastServerInTopologyCacheGroup(t *testing.T) {
 	if len(cgs) != expectedLength {
 		t.Fatalf("expected %d cachegroup with hostname %s, received %d cachegroups", expectedLength, moveToCacheGroup, len(cgs))
 	}
+
+	_, _, err = TOSession.UpdateServerByID(*server.ID, server)
+	if err != nil {
+		t.Fatalf("error updating server with hostname %s without moving it to a different cachegroup: %s", *server.HostName, err.Error())
+	}
+
 	*server.CachegroupID = *cgs[0].ID
 	_, _, err = TOSession.UpdateServerByID(*server.ID, server)
 	if err == nil {
-		t.Fatalf("expected an error moving server with id %d to a different cachegroup, received no error", *server.ID)
+		t.Fatalf("expected an error moving server with id %s to a different cachegroup, received no error", *server.HostName)
 	}
 	if reqInf.StatusCode < http.StatusBadRequest || reqInf.StatusCode >= http.StatusInternalServerError {
 		t.Fatalf("expected a 400-level error moving server with id %d to a different cachegroup, got status code %d: %s", *server.ID, reqInf.StatusCode, err.Error())

--- a/traffic_ops/testing/api/v3/tc-fixtures.json
+++ b/traffic_ops/testing/api/v3/tc-fixtures.json
@@ -3599,7 +3599,7 @@
             "cachegroup": "topology-mid-cg-01",
             "cdnName": "cdn2",
             "domainName": "kabletown.net",
-            "hostName": "midInTopologyEdgeCg01",
+            "hostName": "midInTopologyMidCg01",
             "httpsPort": 443,
             "interfaces": [
                 {
@@ -3633,7 +3633,7 @@
             "cachegroup": "topology-mid-cg-02",
             "cdnName": "cdn2",
             "domainName": "kabletown.net",
-            "hostName": "midInTopologyEdgeCg02",
+            "hostName": "midInTopologyMidCg02",
             "httpsPort": 443,
             "interfaces": [
                 {
@@ -3667,7 +3667,7 @@
             "cachegroup": "topology-mid-cg-03",
             "cdnName": "cdn2",
             "domainName": "kabletown.net",
-            "hostName": "midInTopologyEdgeCg03",
+            "hostName": "midInTopologyMidCg03",
             "httpsPort": 443,
             "interfaces": [
                 {
@@ -3701,7 +3701,7 @@
             "cachegroup": "topology-mid-cg-04",
             "cdnName": "cdn2",
             "domainName": "kabletown.net",
-            "hostName": "midInTopologyEdgeCg04",
+            "hostName": "midInTopologyMidCg04",
             "httpsPort": 443,
             "interfaces": [
                 {
@@ -3735,7 +3735,7 @@
             "cachegroup": "topology-mid-cg-05",
             "cdnName": "cdn2",
             "domainName": "kabletown.net",
-            "hostName": "midInTopologyEdgeCg05",
+            "hostName": "midInTopologyMidCg05",
             "httpsPort": 443,
             "interfaces": [
                 {
@@ -3769,7 +3769,7 @@
             "cachegroup": "topology-mid-cg-06",
             "cdnName": "cdn2",
             "domainName": "kabletown.net",
-            "hostName": "midInTopologyEdgeCg06",
+            "hostName": "midInTopologyMidCg06",
             "httpsPort": 443,
             "interfaces": [
                 {
@@ -3803,7 +3803,7 @@
             "cachegroup": "topology-mid-cg-07",
             "cdnName": "cdn2",
             "domainName": "kabletown.net",
-            "hostName": "midInTopologyEdgeCg07",
+            "hostName": "midInTopologyMidCg07",
             "httpsPort": 443,
             "interfaces": [
                 {

--- a/traffic_ops/traffic_ops_golang/server/servers.go
+++ b/traffic_ops/traffic_ops_golang/server/servers.go
@@ -1226,9 +1226,11 @@ func Update(w http.ResponseWriter, r *http.Request) {
 
 		cacheGroupIds := []int{*origSer[0].CachegroupID}
 		serverIds := []int{*origSer[0].ID}
-		if err := topology.CheckForEmptyCacheGroups(inf.Tx, cacheGroupIds, true, serverIds); err != nil {
-			api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("server is the last one in its cachegroup, which is used by a topology, so it cannot be moved to another cachegroup: "+err.Error()), nil)
-			return
+		if *origSer[0].CachegroupID != *newServer.CachegroupID {
+			if err = topology.CheckForEmptyCacheGroups(inf.Tx, cacheGroupIds, true, serverIds); err != nil {
+				api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("server is the last one in its cachegroup, which is used by a topology, so it cannot be moved to another cachegroup: "+err.Error()), nil)
+				return
+			}
 		}
 
 		server, err = newServer.ToServerV2()

